### DISCUSSION TITLE
Support for custom functions in sql queries?

### DISCUSSION THREAD

**Discussion by jcalven** [simolus3/drift#3265] [Q&A] [answered]
Hi! Does Drift support calling custom dart functions from sql queries, like in the [sqlite3](https://pub.dev/documentation/sqlite3/latest/sqlite3.wasm/CommonDatabase/createFunction.html) package?

**Comment by simolus3** [comment, marked as answer]
Indirectly, yes! You can register custom functions to the connection drift is opening with the `setup` parameter on `NativeDatabase`. Then, once you have a custom function, you can call it in drift by creating a [`FunctionCallExpression`](https://drift.simonbinder.eu/api/drift/functioncallexpression-class) which can be added as a column to evaluate it.

**Reply by jcalven** [reply]
Thank you very much! I will try this.
